### ISSUE TITLE
patterns: general patterns vs specific ones

### ISSUE THREAD

**Issue by matryer** [grafana/explore-logs#352]
Sometimes we extract a pattern that is a superset of other patterns.

For example, the general:

```
level=err msg=<_>
```

might cover many more specific patterns:

```
level=err msg="not found" resource=<_> userID=<_>
level=err msg="conflict" resource=<_> userID=<_>
level=err msg="bad request" resource=<_> userID=<_>
```

* This can happen if multiple streams are working on similar data.

If this is the case, we would ideally hide the general one in favour of the more specific patterns.

---

Ideas:

* This cannot always be known at ingest time, it may need to be a background task or something that runs at query time

**Comment by benclive** [comment]
I completely agree with this - we shouldn't really have patterns that are too generic. 
But on the other hand, we can't just blindly choose the most specific ones because too specific => less useful. There needs to be a middle ground here

To re-use the example above, sometimes you can end up with many patterns like:
```
level=err msg="conflict" resource=/somepath userID=<_>
level=err msg="conflict" resource=/someotherpath userID=<_>
level=err msg="conflict" resource=/veryspecific/path/deep/down?user=ben&view=manual userID=<_>
level=err msg="conflict" resource=/pathA userID=<_>
level=err msg="conflict" resource=/pathB userID=<_>
...
level=err msg="conflict" resource=/pathZ userID=<_>
```

Most of the time these patterns are not useful, other times they are extremely useful (e.g. if one resource is having problems).
I'm starting to wonder if it's possible to (efficiently) allow the user to move up and down the granularity of patterns - I'd only imagine this working if we can do it quickly though - as fast as moving a slider up and down.

**Comment by matryer** [comment]
@benclive that's an interesting idea - zooming in and out of patterns makes a kind of sense. You zoom out, you see things grouped more. You zoom in and you get more granularity. Would you end up doing this at ingest time and just have two modes to switch between? Or is there a way to query the Drain tree with parameters?
